src/XRP/serializer.ts

```diff
@@ -309,6 +326,56 @@ const serializer = {
     }
   },
 
+  /**
+   * Convert an OfferSequence to a JSON representation.
+   *
+   * @param offerSequence - The OfferSequence to convert.
+   * @returns The OfferSequence as JSON.
+   */
+  offerSequenceToJSON(offerSequence: OfferSequence): OfferSequenceJSON {
+    return offerSequence.getValue()
+  },
+
+  /**
+   * Convert an Owner to a JSON representation.
+   *
+   * @param owner - The Owner to convert.
+   * @returns The Owner as JSON.
+   */
+  ownerToJSON(owner: Owner): OwnerJSON | undefined {
+    return owner.getValue()?.getAddress()
+  },
+
+  /**
+   * Convert an EscrowCancel to a JSON representation.
+   *
+   * @param escrowCancel - The EscrowCancel to convert.
+   * @returns The EscrowCancel as JSON.
+   */
+  escrowCancelToJSON(escrowCancel: EscrowCancel): EscrowCancelJSON | undefined {
+    const offerSequence = escrowCancel.getOfferSequence()
+    if (!offerSequence) {
+      return undefined
+    }
+    const offerSequenceJSON = this.offerSequenceToJSON(offerSequence)
+
+    const owner = escrowCancel.getOwner()
+    if (!owner) {
+      return undefined
+    }
+
+    const ownerJSON = this.ownerToJSON(owner)
+    if (!ownerJSON) {
+      return undefined
```

Comment:
is this case not covered by my `malformed owner` test below? i suspect i am missing something